MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -534,8 +560,16 @@ public string GetInputArgument(EncodingJobInfo state, EncodingOptions encodingOp
                 }
 
                 if (state.IsVideoRequest
-                    && (string.Equals(encodingOptions.HardwareAccelerationType, "nvenc", StringComparison.OrdinalIgnoreCase) && isNvdecHevcDecoder || isSwDecoder)
-                        || (string.Equals(encodingOptions.HardwareAccelerationType, "amf", StringComparison.OrdinalIgnoreCase) && isD3d11vaDecoder || isSwDecoder))
+                    && string.Equals(encodingOptions.HardwareAccelerationType, "nvenc", StringComparison.OrdinalIgnoreCase)
+                    && isNvdecDecoder)
+                {
+                    arg.Append("-hwaccel_output_format cuda ");
+                }
+
+                if ((state.IsVideoRequest
+                    && string.Equals(encodingOptions.HardwareAccelerationType, "nvenc", StringComparison.OrdinalIgnoreCase)
+                    && (isNvdecDecoder || isCuvidHevcDecoder || isSwDecoder))
+                        || (string.Equals(encodingOptions.HardwareAccelerationType, "amf", StringComparison.OrdinalIgnoreCase) && (isD3d11vaDecoder || isSwDecoder)))
```

Comment:
```suggestion
                if (state.IsVideoRequest
                    && ((string.Equals(encodingOptions.HardwareAccelerationType, "nvenc", StringComparison.OrdinalIgnoreCase)
                         && (isNvdecDecoder || isCuvidHevcDecoder || isSwDecoder))
                        || (string.Equals(encodingOptions.HardwareAccelerationType, "amf", StringComparison.OrdinalIgnoreCase) 
                            && (isD3d11vaDecoder || isSwDecoder))))
```

I think I suggested incorrect parenthesis, please confirm this suggestion gives the intended result

Reply:
It's time to rewrite the EncodingHelper to avoid such hassle.
HWA methods/pipelines should be separated into different classes.